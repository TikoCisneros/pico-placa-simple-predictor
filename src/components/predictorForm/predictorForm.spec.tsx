import React from 'react';
import { render } from '@testing-library/react';
import PredictorForm from './predictorForm';
import PickersProvider from '../providers/pickersProvider';

describe('PredictorForm test', () => {
  it('should renders correctly', () => {
    const props = {
      licensePlate: '',
      onLicensePlateChange: jest.fn(),
      dateTime: new Date(),
      onDateTimeSelect: jest.fn(),
    };
    const { asFragment } = render(
      <PickersProvider>
        <PredictorForm {...props} />
      </PickersProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
