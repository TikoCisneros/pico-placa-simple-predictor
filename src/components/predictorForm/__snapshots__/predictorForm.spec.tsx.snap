// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PredictorForm test should renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root sc-bdnxRM hppKkw MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root sc-gtsrHT ckibhS"
    >
      <img
        alt="logo"
        class="sc-dlnjwi DaiG"
        src="trafficJam.svg"
      />
      <h4
        class="MuiTypography-root MuiTypography-h4 MuiTypography-colorPrimary MuiTypography-gutterBottom"
      >
        Vehicle circulation predictor
      </h4>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
          data-shrink="true"
        >
          License plate
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            placeholder="e.g. AXC0123"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
            >
              <span>
                License plate
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        format="MMMM Do hh:mm a"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
          data-shrink="true"
        >
          Date and time
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
            readonly=""
            type="text"
            value="June 12th 11:17 pm"
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 2v3H4V5h16zM4 21V10h16v11H4z"
                  />
                  <path
                    d="M4 5.01h16V8H4z"
                    opacity=".3"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
            >
              <span>
                Date and time
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
        style="margin-top: 15px;"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Can I drive?
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
        style="margin-top: 15px;"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Do you know the restrictions yet?
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
